tests/test_task.py

```diff
@@ -184,19 +235,31 @@ def test_reproduce_experiment(
     trials = task.generate_trials(ntrials=ntrials, random_seed=random_seed)
     task_repro = Task(**trials["task_settings"])
     trials_repro = task_repro.generate_trials(trials["ntrials"], trials["random_seed"])
+    trial_indices = range(trials["ntrials"])
 
     # Check that the output is the same
     assert task == task_repro
     for x, y in trials.items():
-        if x != "phases":  # tested separately because it's a dict of numpy arrays
-            assert np.array_equal(y, trials_repro[x])
-    assert all(np.array_equal(trials["phases"][key], trials_repro["phases"][key]) for key in trials["phases"])
+        if type(y) is not np.ndarray:
+            # task_settings, ntrials, random_seed
+            assert y == trials_repro[x]
+        elif type(y[0]) is np.ndarray:
```

Comment:
is there a third option? If so it should be tested, if not, this should just state else so that if something is added in the future, it will fail and we will catch it.

Reply:
There are other two options, which are tested below (`"phases"` and `"modality_seq"`). I added an assert for checking that all the keys of `trials` are tested, so we will know if we forget to add anything there.